### PR TITLE
Add new properties to Dispute model

### PR DESCRIPTION
**Pull request by igroomgrim**
Added 4 new properties to Dispute model

- `transaction`
- `reason_code`
- `reason_message`
- `closed_at`

Note: For `documents` field will add in new PR.

### CODE REVIEW

File: src/main/java/co/omise/models/Dispute.java [hunk 1]
```diff
@@ -76,6 +84,38 @@ public void setMetadata(Map<String, Object> metadata) {
         this.metadata = metadata;
     }
 
+    public void setReasonCode(DisputeReasonCode reasonCode) {
```
Comment (Namiii) [inline src/main/java/co/omise/models/Dispute.java]: Do we need all these `getters` and `setters`?

File: src/main/java/co/omise/models/DisputeReasonCode.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package co.omise.models;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public enum DisputeReasonCode {
+    @JsonProperty("cancelled_recurring_transaction")cancelledRecurringTransaction,
```
Comment (Namiii) [inline src/main/java/co/omise/models/DisputeReasonCode.java]: This work like this out of the box? Don't we need to add custom logic to the serializer?

File: src/test/resources/testdata/fixtures/api.omise.co/disputes/dspt_test_5089off452g5m5te7xs-get.json [hunk 3]
```diff
@@ -12,5 +12,9 @@
   "metadata": {
     "description": "DESCRIPTION",
     "invoice_id": "inv_N1ayTWJ2FV"
-  }
+  },
+  "transaction": "trxn_test_59bqoovjrgehphloams",
+  "reason_code": "goods_or_services_not_provided",
+  "reason_message": "Services not provided or Merchandise not received",
+  "closed_at": null
```
Comment (nuxzero) [inline src/test/resources/testdata/fixtures/api.omise.co/disputes/dspt_test_5089off452g5m5te7xs-get.json]: Should we add some date to test the date can serializable?

Comment (igroomgrim) [reply src/test/resources/testdata/fixtures/api.omise.co/disputes/dspt_test_5089off452g5m5te7xs-get.json]: @nuxzero Sure, Added 🙇

File: src/main/java/co/omise/models/DisputeReasonCode.java (hunk 2 above)
Comment (igroomgrim) [reply src/main/java/co/omise/models/DisputeReasonCode.java]: @Namiii In my opinion, I think we need it. That makes it easy to use but after re-think, should we just let it be String and let Customer map it by themselves? 🤔

Comment (Namiii) [reply src/main/java/co/omise/models/DisputeReasonCode.java]: No if it works fine, we should leave it like this. I was just wondering

Comment (igroomgrim) [reply src/main/java/co/omise/models/DisputeReasonCode.java]: Got it. 😄 👌

**Pull request merged**